R/query_search.R

```diff
@@ -231,8 +231,14 @@ query_eval_test_binary <- function(op, a, b) {
   if (length(a) == 0 || length(b) == 0) {
     return(logical(0))
   }
-  vlapply(Map(function(a, b) !is.null(a) && !is.null(b) && op(a, b),
-              a, b, USE.NAMES = FALSE),
+  is_equal <- function(a, b) {
+    !is.null(a) && !is.null(b) &&
+      (is.numeric(a) && is.numeric(b) ||
+         is.logical(a) && is.logical(b) ||
+         is.character(a) && is.character(b)) &&
```

Comment:
`storage.mode(a) == storage.mode(b)
```suggestion
      storage.mode(a) == storage.mode(b) &&
```

Reply:
Is there a way we can make this work for integer and numeric types?

```
> storage.mode(1L) == storage.mode(1)
[1] FALSE
```

Reply:
good point. This should work:

```
same_type <- function(a, b) {
  (is.numeric(a) && is.numeric(b)) || (storage.mode(a) == storage.mode(b))
}
```